pandas/tests/arrays/masked/test_arithmetic.py

```diff
@@ -55,15 +55,15 @@ def test_array_scalar_like_equivalence(data, all_arithmetic_operators):
     scalar_array = pd.array([scalar] * len(data), dtype=data.dtype)
 
     # TODO also add len-1 array (np.array([scalar], dtype=data.dtype.numpy_dtype))
-    for scalar in [scalar, data.dtype.type(scalar)]:
+    for val in [scalar, data.dtype.type(scalar)]:
```

Comment:
What rule is this? 

Not sure I like this change (or if its worth the churn).

Reply:
https://docs.astral.sh/ruff/rules/loop-variable-overrides-iterator/

I think this could be useful generally. In the file above (`pandas/io/json/_normalize.py`), if `v` needed to be used after looping it would no longer be the original object it would be an object element

Reply:
My bad, wasn't paying too close attention.

I guess a more descriptive name would be nicer, but I'm fine with it as it is.